model_hub/examples/huggingface/token-classification/ner_utils.py

```diff
@@ -0,0 +1,155 @@
+"""
+These functions extract parts of the code from the named-entity recognition example
+in huggingface transformers.  The source example code changes rapidly due to frequent refactorings
+but the main structure should be similar.
+See the original example at:
+https://github.com/huggingface/transformers/blob/master/examples/token-classification/run_ner.py
+
+Note this code is separated from ner_trial.py since it is not Determined-specific code and
+is necessary even if implementing outside of Determined.
+"""
+from typing import Any, Dict, List, Union
+
+import attrdict
+import datasets as hf_datasets
+import numpy as np
+import torch
+from seqeval import metrics as seq_metrics
+
+from model_hub import huggingface as hf
+from model_hub import utils
+
+# The functions below are largely derived from the associated transformers example.
```

Comment:
praise: Thank you for carefully handling the derived portions of the code.

nit: since this is all legalese, and laywers can't be trusted to figure the right thing out on their own, can you be more specific than "the functions" and more specific than "the associated example"?

Reply:
:ok_hand: